lib/xboxrt/libc_extensions/stdlib_ext_.c

```diff
@@ -0,0 +1,66 @@
+#include <assert.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <threads.h>
+#include <windows.h>
+#include <xboxkrnl/xboxkrnl.h>
+
+once_flag init_flag = ONCE_FLAG_INIT;
+mtx_t rand_mutex;
+unsigned char rand_buffer[116];
+
+void update_rand_buffer (void)
+{
+    DWORD tickCount = GetTickCount();
+    XcSHAUpdate(rand_buffer, (PUCHAR)&tickCount, sizeof(tickCount));
+
+    LARGE_INTEGER performanceCounter;
+    QueryPerformanceCounter(&performanceCounter);
+    XcSHAUpdate(rand_buffer, (PUCHAR)&performanceCounter, sizeof(performanceCounter));
+}
+
+void init_rand_s (void)
+{
+    mtx_init(&rand_mutex, mtx_plain);
+
+    XcSHAInit(rand_buffer);
+
+    // Hash the EEPROM
+    unsigned char eeprom_buffer[256];
+    ULONG bytes_read;
+    ULONG eeprom_type;
+    ExQueryNonVolatileSetting(0xFFFF, &eeprom_type, eeprom_buffer, 256, &bytes_read);
+    assert(bytes_read == 256);
+    XcSHAUpdate(rand_buffer, eeprom_buffer, 256);
+
+    update_rand_buffer();
+}
+
+int rand_s (unsigned int *randomValue)
```

Comment:
May want to leave a comment that this probably isn't a very cryptographically secure random number generation as the MS docs indicate

Reply:
Done.